plugins/backend/catalog-backend-module-aws/src/providers/AWSDynamoDbTableDataProvider.ts

```diff
@@ -0,0 +1,170 @@
+  private readonly tableDataConfig: DdbTableDataConfigOptions;
+  private readonly logger: winston.Logger;
+
+  private connection?: EntityProviderConnection;
+
+  static fromConfig(config: Config, options: { logger: winston.Logger }) {
+    return new AWSDynamoDbTableDataProvider(
+      config.getString('accountId'),
+      config.getString('roleArn'),
+      config.get<DdbTableDataConfigOptions>('dynamodbTableData'),
+      options.logger,
+    );
+  }
+
+  constructor(
+    accountId: string,
+    roleArn: string,
+    options: DdbTableDataConfigOptions,
+    logger: winston.Logger,
+  ) {
+    this.accountId = accountId;
+    this.roleArn = roleArn;
+    this.tableDataConfig = options;
+    this.logger = logger;
+  }
+
+  getProviderName(): string {
+    return `aws-dynamo-db-table-${this.accountId}-${this.tableDataConfig.tableName}`;
+  }
+
+  async connect(connection: EntityProviderConnection): Promise<void> {
+    this.connection = connection;
+  }
+
+  async run(): Promise<void> {
+    if (!this.connection) {
+      throw new Error('Not initialized');
+    }
+
+    const credentials = fromTemporaryCredentials({
+      params: { RoleArn: this.roleArn },
+    });
+
+    const ddb = DynamoDBDocument.from(
+      new DynamoDBClient({
+        credentials,
+      }),
+    );
+    const dynamoDBClient = new DynamoDB({ credentials });
+
+    const account = await new STS({ credentials }).getCallerIdentity({});
+
+    const defaultAnnotations: { [name: string]: string } = {
+      [ANNOTATION_LOCATION]: `${this.getProviderName()}:${this.roleArn}`,
+      [ANNOTATION_ORIGIN_LOCATION]: `${this.getProviderName()}:${this.roleArn}`,
+      'amazon.com/dynamodb-table-name': this.tableDataConfig.tableName,
+    };
+
+    if (account.Account) {
+      defaultAnnotations['amazon.com/account-id'] = account.Account;
+    }
+
+    this.logger.info(`Querying table ${this.tableDataConfig.tableName}`);
+
+    try {
+      const tableRows = await ddb.scan({
+        TableName: this.tableDataConfig.tableName,
+      });
+      const tableDescription = await dynamoDBClient.describeTable({
+        TableName: this.tableDataConfig.tableName,
+      });
+      const idColumn =
+        this.tableDataConfig.identifierColumn ||
+        tableDescription.Table?.KeySchema?.[0].AttributeName;
+      if (!idColumn) {
+        throw new Error(
+          'No identifier column defined. Unable to construct entities.',
+        );
+      }
+      const tableArn = tableDescription?.Table?.TableArn;
+      const entities =
+        tableRows.Items?.map(row => {
+          const o = {
+            kind: 'Component',
+            apiVersion: 'backstage.io/v1beta1',
+            metadata: {
+              annotations: {
+                ...defaultAnnotations,
+                ...(tableArn ? { 'amazon.com/dynamo-db-table': tableArn } : {}),
+              },
+              name: row[idColumn],
+            },
+            spec: {
+              owner: 'unknown',
```

Comment:
Could this possibly be the account id?
```suggestion
              owner: this.accountId,
```

Reply:
Good idea :+1: